### ISSUE TITLE
contract schema allows required fields that are not defined in the schema, breaking the contract

### ISSUE THREAD

**Issue by dashameter** [dashevo/js-dpp#183]
## Expected Behavior
When registering a contract containing required fields that are not defined in the schema itself, the state transition should be rejected.

## Current Behavior
The state transition is accepted, breaking the contract.

## Steps to Reproduce (for bugs)
1.) Load 'valid' contractId `5g7bvdbxGGY7czP26cZ91Cdu9DwqpvaCn7QKJxCvb1Tq` at http://console.dashevo.io/#/platform
2.) Expand 'view contract'
3.) Try submitting a `note` document: `{ "message": "darn gnomes"}`
4.) Register example 'valid' contract below.
```
{
 "$id": "5g7bvdbxGGY7czP26cZ91Cdu9DwqpvaCn7QKJxCvb1Tq",
 "$schema": "https://schema.dash.org/dpp-0-4-0/meta/data-contract",
 "ownerId": "FbXKh6tay1RuGW2G41MRG2XtWUXokwat5M5nCrU7U2kD",
 "documents": {
  "note": {
   "required": [
    "sexyGnome"
   ],
   "properties": {
    "message": {
     "type": "string"
    }
   },
   "additionalProperties": false
  }
 },
 "definitions": {}
}
```

## Your Environment
    "dash": "^3.0.2"

**Comment by shumkov** [comment]
Yeah, JSON Schema allows that.

**Comment by shumkov** [comment]
Fixed in DPP v0.20
